.ci/test.sh

```diff
@@ -51,10 +51,13 @@ if [[ $TRAVIS == "true" ]] && [[ $TASK == "check-docs" ]]; then
 fi
 
 if [[ $TASK == "lint" ]]; then
+    sudo apt-get update
     conda install -q -y -n $CONDA_ENV \
         pycodestyle \
-        pydocstyle \
-        r-lintr
+        pydocstyle
+    conda install -q -y -n $CONDA_ENV -c conda-forge \
+        r-lintr==2.0.0
+    Rscript -e "install.packages('stringi', repos = 'http://cran.rstudio.com')"
```

Comment:
It takes soo much time! Is it the only available fix? Maybe it will be faster to update `lintr` from CRAN?

Reply:
Oh no! I think we should stick to conda's environment, since it's significantly faster than ordinary one and suits to our purpose (just linting) very well.
Sorry for my ambiguous comment above! I just meant to try `Rscript -e "install.packages('lintr', repos = 'http://cran.rstudio.com')"` to see whether it's faster than reinstalling `stringi` or not.

Reply:
ah I see, ok. Yes I think I misinterpreted what you were saying. I'll change it back and try.

Reply:
Looks like linting task took [almost 7 minutes to run](https://travis-ci.org/microsoft/LightGBM/jobs/607877757?utm_medium=notification&utm_source=github_status) still with this approach (using alternative R setup)

![image](https://user-images.githubusercontent.com/7608904/68249354-482c4d00-ffe4-11e9-9079-07ce05421ead.png)

Reply:
creating an R environment from `conda` and then `install.packages('lintr')` from CRAN, the `lint` task ran for 8m38s and then failed with

```
ERROR: dependency ‘curl’ is not available for package ‘httr’
```

I could add an `install.packages('curl')` but that would make it run even longer than 8 minutes.

I decided to summarize all the attempts here (and add a few new ideas):

1. ([-failure-](https://travis-ci.org/microsoft/LightGBM/builds/605269042?utm_medium=notification&utm_source=github_status)) R env from conda, `lintr` from conda-forge
2. ([success](https://travis-ci.org/microsoft/LightGBM/jobs/606612630?utm_medium=notification&utm_source=github_status)) R env from conda, `lintr` from conda-forge, `install.packages('stringi')`: 6m9s
3. ([success](https://travis-ci.org/microsoft/LightGBM/builds/607877752?utm_medium=notification&utm_source=github_status)) R env from CRAN like #2530, `install.packages('lintr')`: 6m48s
4. ([-failure-](https://travis-ci.org/microsoft/LightGBM/builds/607939450?utm_medium=notification&utm_source=github_status)) R env from conda, `install.packages('lintr')`: 8+ minutes
5. ([-failure-](https://travis-ci.org/microsoft/LightGBM/jobs/607962730?utm_medium=notification&utm_source=github_status)) R env from conda, `lintr` from conda-forge, re-installing `stringi` from default conda channels: failed
6. ([success](https://travis-ci.org/microsoft/LightGBM/jobs/607963803?utm_medium=notification&utm_source=github_status)) R env from conda, installing `stringi` from conda first, then `lintr` from conda-forge: 1m57s

So I think we should go with installing `stringi` explicitly, THEN `lintr`! This works and runs in under 2 minutes 🎉🎉 🎉